### PR TITLE
Improve coding conventions

### PR DESCRIPTION
**Pull request by drwells**
Quick followup to #8819: I also fixed a few minor things in the coding conventions document while I was there.

### CODE REVIEW

File: doc/doxygen/headers/coding_conventions.h [hunk 1]
```diff
@@ -119,7 +119,33 @@ executable.
...
+    int a = 42;
+    int *b = nullptr;
+  };
+@endcode
+  write
+@code
+  class Foo
+  {
+    Foo();
+  };
```
Comment (bangerth) [inline doc/doxygen/headers/coding_conventions.h]: add the member variables here, without their initializers

File: doc/doxygen/headers/coding_conventions.h [hunk 2]
```diff
@@ -119,7 +119,33 @@ executable.
   then functions.
   <br>
   Exceptions shall be declared at the end of the public section
-  before the non-public sections start.</li>
+  before the non-public sections start.
+  <br>
+  We do not use the C++11-style class member initialization for member variables
+  that are neither <code>static const</code> nor <code>static constexpr</code>;
+  i.e., instead of
```
Comment (bangerth) [inline doc/doxygen/headers/coding_conventions.h]: do we provide rationales for the other rules, or do we just say that that's what it is? if the former, feel free to add a sentence explaining why.

File: doc/doxygen/headers/coding_conventions.h (hunk 1 above)
Comment (drwells) [reply doc/doxygen/headers/coding_conventions.h]: whoops.

File: doc/doxygen/headers/coding_conventions.h (hunk 2 above)
Comment (drwells) [reply doc/doxygen/headers/coding_conventions.h]: We just enumerate the rules here: nothing else under 'style issues' gives an explanation.

**Pull request merged**